hooks/useOrders.ts

```diff
@@ -0,0 +1,72 @@
+import { useOrderHistory } from "@components/user/history/OrderHistoryProvider";
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react";
+import { useQuery, UseQueryResult } from "@tanstack/react-query";
+import { Order } from "models/Order";
+import { BoughtProduct, CartProduct } from "models/Product";
+import { Dispatch, SetStateAction } from "react";
+
+export const useOrders = (): UseQueryResult<Order[]> => {
+  const supabase = useSupabaseClient();
+  const user = useUser();
+  const { setOrders, setSelectedOrder } = useOrderHistory();
+
+  return useQuery(
+    ["orders"],
+    async () => {
+      const { data, error } = await supabase
+        .from("orders")
+        .select()
+        .eq("user_id", user!.id)
+        .order("id", { ascending: false });
+
+      if (error) throw error;
+      if (data) return data as Order[];
+    },
+    {
+      enabled: !!user,
+      onSuccess: (orders) => {
+        setOrders(orders);
+        setSelectedOrder(orders[0]);
+      },
+    }
+  );
+};
+
+export const useBoughtProducts = (
+  setBoughtProducts: Dispatch<SetStateAction<CartProduct[]>>
+): UseQueryResult<CartProduct[]> => {
+  const supabase = useSupabaseClient();
+  const { selectedOrder } = useOrderHistory();
+
+  return useQuery(
+    ["boughtProducts", selectedOrder],
+    async () => {
+      const { data, error } = await supabase
+        .from("bought_products")
+        .select()
+        .or(selectedOrder!.bought_products_id!.map((id) => `id.eq.${id}`).join(","));
+      const boughtProducts = data as BoughtProduct[];
+
+      if (error) throw error;
+      if (boughtProducts) {
+        const getProducts = async () => {
+          const { data, error } = await supabase
+            .from("products")
+            .select()
+            .or(boughtProducts.map((product) => `id.eq.${product.product_id}`).join(","));
```

Comment:
There is a lot of mapping happening here. 
- if you are making 2 queries, in order to fetch a `mapped` version of your data (because they life in 2 different tables) consider using `Views`. It's a feature provided by Postgress which Supababse supports :)

Reply:
Ain't going to mess up anymore with these queries. I tried to fetch those in multiple ways and couldn't bring more than 1 item at a time.